src/components/main/GiftsList.tsx

```diff
@@ -0,0 +1,74 @@
+import styled from "@emotion/styled";
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router";
+import GiftItem from "./GiftItem";
+import { ROUTE_PATH } from "@/routes/paths";
+import type { Gift } from "@/types/gift";
+
+type GiftsListProps = {
+  gifts: Gift[];
+};
+
+const GiftsList = ({ gifts }: GiftsListProps) => {
+  const navigate = useNavigate();
+  const [showMore, setShowMore] = useState(false);
+
+  const handleGiftClick = (giftId: number) => {
+    navigate(`${ROUTE_PATH.ORDER.replace(":id", giftId.toString())}`);
+  };
+
+  useEffect(() => {
+    setShowMore(false);
+  }, [gifts]);
+
+  const visibleGifts = showMore ? gifts : gifts.slice(0, 6);
+
+  return (
+    <>
+      <GiftsGrid>
+        {visibleGifts.map((gift, index) => (
+          <GiftItem
+            key={gift.id}
+            gift={gift}
+            rank={index + 1}
+            as="button"
+            type="button"
+            onClick={() => handleGiftClick(gift.id)}
+          />
```

Comment:
앞선 리뷰의 연장선인데요,

```suggestion
          <GiftItem
            key={gift.id}
            {...gift}
            rank={index + 1}
            as="button"
            type="button"
            onClick={() => handleGiftClick(gift.id)}
          />
```
요로코롬 gift를 해체해서 사용해주면 어떨까 싶어요!

Reply:
객체 구조 분할을 사용하면 단순한 형태로 props의 depth를 낮춰서 전달할 수 있겠네요.